### ISSUE TITLE
Wayland Support

### ISSUE THREAD

**Issue by barfin** [NVIDIA/nvidia-settings#69]
nvidia-settings is an essential tool for linux users to tweak their graphics card under a linux system,
not a 'nice-to-have' program, until this tool gets ported to Wayland I won't be able to use my graphics card the way i want so I'll stick to xorg

**Comment by xo1n** [comment]
Nvidia, please make Wayland (Sway) support in your Drivers and in nvidia-settings!

**Comment by freevatar** [comment]
My vote for Wayland support as well.

**Comment by Mondrethos** [comment]
This is really important.

**Comment by ivanfiorentini** [comment]
nvidia-settings  is mandatory to control color correction of my monitor.
Please **do support  nvidia-settings for wayland** !

**Comment by CleoMenezes** [comment]
I don't understand why this issue was closed, since it still doesn't work.

**Comment by barfin** [comment]
> I don't understand why this issue was closed, since it still doesn't work.

It isnt

**Comment by takosalad** [comment]
Yes please, also, I need NVidia Reflex on Linux, playing Overwatch there heavily.

**Comment by nulm** [comment]
Needed

**Comment by tfiskgul** [comment]
Adding my voice to this request. Please support Wayland.

**Comment by ifohancroft** [comment]
I am also in-favor of this.

**Comment by CleoMenezes** [comment]
All in one petition!

**Comment by phodina** [comment]
Please add wayland support :-)

**Comment by pukmajster** [comment]
I hear Wayland is cool.

**Comment by CleoMenezesJr** [comment]
Well, we're finally getting something.
![Captura de tela de 2022-05-08 19-12-42](https://user-images.githubusercontent.com/54215258/167319760-6f682b31-a3e7-4712-bd3d-95412278cca3.png)

**Comment by pukmajster** [comment]
It's a step in the right direction for sure

**Comment by sachnr** [comment]
An option to change dithering setting like the one in xorg would be appreciated. The default ones causes weird artifacts currently no option to change it.

**Comment by TomPed** [comment]
Happy to see others talking about this 😄 . @pukmajster, yeah it's a great step forward!

**Comment by gilvbp** [comment]
Hi everyone, just to remind you all,  we need nvidia-settings with the same functionality just like with the x11 version. Thanks.

**Comment by ex1tium** [comment]
Any progress on this?

**Comment by mamaraddio** [comment]
We all are still asking for Wayland support.
**NVIDIA come on!**

**Comment by alexknop** [comment]
PowerMizer settings aren't there when I use Wayland, but are when I use Xorg. Please work to convert nvidia-settings to Wayland.

**Comment by PlatinumLucario** [comment]
Is anyone here prepared to make a fork of nvidia-settings and bind all the settings to their respective APIs within a Wayland session, so that all the Wayland compositors will show all the settings in the nvidia-settings application?

**Comment by Samir2zk** [comment]
Please, NVIDIA for past cards as well. We need Wayland support. Open source just like the new series the portion of the drivers.

**Comment by gardotd426** [comment]
> Is anyone here prepared to make a fork of nvidia-settings and bind all the settings to their respective APIs within a Wayland session, so that all the Wayland compositors will show all the settings in the nvidia-settings application?

I actually don't think this is possible, Nvidia Settings uses libxnvctrl for all hardware control. libxnvctrl is an X extension and cannot be modified to work in Wayland no matter what. And Nvidia don't open up their hardware for user-control in sysfs, so you can't just force it. It's actually not possible. 

I'm a bit baffled as to why Nvidia aren't even acknowledging this, though. They fully know that Wayland is coming and is going to be the default for pretty much every distro very soon, and they HAVE to allow users to modify fan curves.

**Comment by hadley8899** [comment]
Up!

**Comment by abishekmuthian** [comment]
After updating to 545 drivers the fans didn't seem to respect the default curves and were running at lower temps. So I enforced custom fan curves using `asusctl` - https://gitlab.com/asus-linux/asusctl ; I understand that this information is not helpful to most, but  I'm writing this comment in case someone with Asus laptops visit here looking for nvidia-settings .

**Comment by Bezalelohim** [comment]
Wayland is the way forward.... Nvidia Support

**Comment by gardotd426** [comment]
You can't add features that require proprietary libraries only available on
X11 like libxnvcntrl. Duh.

**Comment by PlatinumLucario** [comment]
Perhaps decompiling the libraries might help give everyone a clue on what new libraries to create and implement based on the APIs in the Wayland protocol?

**Comment by Bezalelohim** [comment]
> Perhaps decompiling the libraries might help give everyone a clue on what new libraries to create and implement based on the APIs in the Wayland protocol?

That would be huge, highly doubt it would happen though, but there is another "from the ground up solution" in the works, [NVK_Vulkan](https://www.phoronix.com/review/nvk-vulkan-performance), sadly it will take a very long time to reach somewhere close to the levels of proprietary drivers and their feature sets, nonetheless it is a good option to have.

**Comment by Augusto7743** [comment]
Wayland in paper is amazing, but in reality is an unfinished protocol ... how are happening problems about adding wayland !
Other problem is some distro dropping support for x11 and so NOT will be possible use "old" hardware in future distro.
Several notebook models and video cards NOT will work in the future ... That's bizarre for Linux if comparing with windows support for "old" hardware.
Need to be created an solution .. not much hope in Nvidia for video cards before GTX 700 series ...
However Nouveau can "fix" it .... but not has support for CUDA, NVENC and good performance.

**Comment by ErrorNoInternet** [comment]
Not exactly sure what you're saying but...

Some distros are dropping X11, but not all of them are. You can always use a different distro. They are also not preventing X11 code from ever running on your computer. You can always download and set up X11 yourself. Linux is the kernel, not the entire operating system. Linux itself (and lots of projects) still have tons of support for old hardware, it's just that "distro people" don't wanna package/polish/set up extremely old legacy code anymore.

Wayland is a protocol, not a hardware feature. I am still running Wayland on ancient laptops from 2012 without any issues. I can do the exact same things on X11 and Wayland (including but not limited to screen sharing, streaming, color management (their displays are so old they look yellow now), multi-monitors, and so much more)

NVIDIA doesn't wanna fully support Wayland in their old drivers, and that's their problem. You can't really compare Windows and Linux when Linux is an open source project full of volunteers, and a multimillion dollar company is choosing to not support it.

**Comment by Augusto7743** [comment]
"Also, why not just use an older distro?"
simply because some old distro has problems installing or compiling kernel new versions.

I wait Ubuntu not remove x11 in future versions so allowing use "old" video cards.
Wayland was added very soon and have done much problems mainly users demanding compatibility to an unfinished protocol.

About windows removing support for "old" hardware .... well ... in windows universe that detail and the terrible security updates (intel cpu losing more than 15 performance) in 2018 are example of as them does actions against users and waiting high profit for big companies.

Other strange detail is Nvidia having an long time to code an fully support for wayland. Perhaps in begin some years back not want wayland to be an protocol used in the world ?

Even so about video card support Nvidia is much generous ... video cards being supporte for more than 10 years and other companies are less of 5 years ...
Also Kepler 600 and 700 GT series has support for wayland if using egl. I never had test it if the performance is good as gbm.

**Comment by cutephoton** [comment]
It's been nothing but problems since 2020ish even when using Xorg. Distributions have bent over backwards trying to accommodate nvidia's tardiness. Today was the day I gave up. I'm done. I've returned to AMD. I did not want to return to AMD, but using wayland and nvidia is quite possibly the worst experience I've had in more than a decade.

**Comment by pukmajster** [comment]
It's weird. For some, nvidia + wayland works really well, but for me it was straight up unusable because chromium based apps kept having seizures and the input lag in games was way too much

**Comment by iddm** [comment]
> It's weird. For some, nvidia + wayland works really well, but for me it was straight up unusable because chromium based apps kept having seizures and the input lag in games was way too much

Have never had an input lag and other issues (except the chromium-based garbage) on 3080, 4090 and 4070. Weird?

**Comment by exalented** [comment]
> It's weird. For some, nvidia + wayland works really well, but for me it was straight up unusable because chromium based apps kept having seizures and the input lag in games was way too much

https://gitlab.freedesktop.org/xorg/xserver/-/issues/1317

Solution needs to be merged into stable. It'll take some time.

Nvidia, your customer base is almost ready to switch to an open solution on this 4 year old, basic issue. Throw us the blob for your closed GPU and let the community handle the rest.

**Comment by SSkizz** [comment]
Would still greatly appreciate the ability to change monitor specific settings, like refresh rate, digital vibrance, and resolution.

**Comment by kbrenneman** [comment]
> Would still greatly appreciate the ability to change monitor specific settings, like refresh rate, digital vibrance, and resolution.

With X11, you've got a standard config file for initial settings, and you've got RANDR for on-the-fly changes. So, you can have a single program that can configure your monitors, and that program will work with any window manager or desktop environment you care to use.

In the *delightfully* fragmented world of Wayland, no such program can exist. All of the monitor settings are up to the compositor, and there's no standard interface or protocol to configure them.

**Comment by ErrorNoInternet** [comment]
> In the *delightfully* fragmented world of Wayland, no such program can exist. All of the monitor settings are up to the compositor, and there's no standard interface or protocol to configure them.

https://wayland.app/protocols/wlr-output-management-unstable-v1

https://sr.ht/~emersion/kanshi/

https://sr.ht/~emersion/wlr-randr/

**Comment by bencardoen** [comment]
On Fedora 38 Wayland + Nvidia leads to tearing, screen artifacts, etc, which go away once you switch to X11, go into nvidia-settings and switch powermizer to max performance. The option isn't available in nvidia-settings + wayland (as noted above). So this isn't just a 'I'd like to configure my GPU on Wayland', it's actually needed to work around driver bugs.

Curiously the issues are only with Chromium based (and thus electron) apps, and Steam. Firefox is fine. In X11 no issues appear. 

Anyone know if the powermizer settings are controllable by another CLI app? Tried nvidia-smi but no luck so far.

**Comment by wassou93** [comment]
Any updates on this ?

**Comment by Augusto7743** [comment]
Why Nvidia not does any action to nvidia settings work in wayland ?
Seriously Nvidia dislike wayland ?

Is possible configure all nvidia settings editing config files under wayland ? works withoout errors ?
Nvidia-smi works in wayland ?

Have an nice week.

**Comment by BugReporterZ** [comment]
Ideally fan speed control, and also importantly the Coolbits functionality for core/memory overclock could be integrated into nvidia-smi, which AFAIK doesn't need Wayland nor X to function.

**Comment by Krovikan** [comment]
Any news for nvidia-settings + Wayland?

The date of driver with ESYNC is near (end of May) and we don't have the configuration tool of Nvidia to switch to Wayland.

Regards

**Comment by pallaswept** [comment]
> nvidia-smi, which AFAIK doesn't need Wayland nor X to function.

It needs X

**Comment by pukmajster** [comment]
Not a whole lot to do without proper access the the driver code itself.

**Comment by Krovikan** [comment]
> I just don't understand how is nvidia-settings not working well under wayland

I understand that is because nvidia-settings is written only for X11 as others programs (like Simple Screen Recorder and others) that works in X11 and in Wayland refuse to launch if is not X11). 

Regards

The reason? I don't know. I am bot a programmer.

**Comment by HackTestes** [comment]
@BlueGoliath

> > @BugReporterZ
> > > Ideally fan speed control, and also importantly the Coolbits functionality for core/memory overclock could be integrated into nvidia-smi, which AFAIK doesn't need Wayland nor X to function.
> > 
> > 
> > Regarding fan speed, I still don't understand how is this still a problem, because NVIDIA is already using the NVML API which doesn't depend on a display server running. Look at the code here: https://github.com/NVIDIA/nvidia-settings/blob/main/src/libXNVCtrlAttributes/NvCtrlAttributesNvml.c#L1578
> > And, I can already report that the NVML functions added to to drivers higher than 520 work under Wayland. I have also been working on an alternative program (just check my repositories for NVML GPU Control) and at least one user reported that he has it working perfectly on his machine (I am using Windows right now, so I can't test it directly).
> > So me and NVIDIA are using the same API functions, I just don't understand how is nvidia-settings not working well under wayland already (it should be).
> > Well, just wanted to share some insight to others reading the issues as this might help someone to figure out the underlying problem.
> 
> Wow how did you figure out all that without having proper access to driver code itself?!?!?

This might sound a bit silly, but the short answer is: I just read the NVML change logs (look for "Changes between v510 and v515" here: https://docs.nvidia.com/deploy/nvml-api/change-log.html#change-log).

Long answer: I was searching for more than a year what kind of APIs NVIDIA has to control their GPUs. I came across NVAPI (Windows only), NVML (used mostly for professional cards and on nvidia-smi utility) and libXNVCtrl (depends on X11). Then I saw the change logs on NVML and I rushed to make my own software to control the GPUs. Tested on my machine and It worked perfectly! So, it was a bunch of documentation reading and some recent testing.

I was simply tired of GeForce Experience (especially after I found out about the online requirement) and wanted to switch to Wayland Linux (which lacked a way to control the GPU's fan). You can also try out my code if you to verify it better (it is super small so others can verify it in about an hour).

By the way, I have also asked for people on Reddit to help me test it under Wayland (I wanted to check if it worked, before I make any post here). Only 2 guys showed up, but they were super helpful in testing it.

**Comment by HackTestes** [comment]
@BlueGoliath

I forgot to mention, I also saw in a GitHub issue a NVIDIA employee talking that they were planning to add many features to NVML, instead of relying on libXNVCtrl (and that was how I learned about the possible importance of the NVML API). I just didn't put this information on the first post, because I couldn't find the issue again (I lost the link).

**Comment by Krovikan** [comment]
> Long answer: I was searching for more than a year what kind of APIs NVIDIA has to control their GPUs. I came across NVAPI (Windows only), NVML (used mostly for professional cards and on nvidia-smi utility) and libXNVCtrl (depends on X11). Then I saw the change logs on NVML and I rushed to make my own software to control the GPUs. Tested on my machine and It worked perfectly! So, it was a bunch of documentation reading and some recent testing.

Your work is very commendable. But we will all agree that the complete configuration module is needed. That is, this:
![imagen](https://github.com/NVIDIA/nvidia-settings/assets/3192501/f790b159-8e20-47f7-b729-a85b3603db1b)

Best regards

**Comment by HackTestes** [comment]
@Krovikan 
> > I just don't understand how is nvidia-settings not working well under wayland
> 
> I understand that is because nvidia-settings is written only for X11 as others programs (like Simple Screen Recorder) that works in X11 and in Wayland refuse to launch if is not X11.
> 
> Regards
> 
> The reason? I don't know. I am not a programmer.

I know it was made for X11, but people can already open it under Wayland. There is even a big post on Reddit from some years ago that showed people celebrating this fact. Here: https://www.reddit.com/r/Fedora/comments/sliye1/and_so_the_day_we_can_open_nvidia_settings_in_the/?rdt=56145

I have also seen many posts and issues from various program that can also open nvidia-settings in Wayland.

When I say that I don't understand the reason is because NVIDIA **updated** nvidia-settings to use NVML (and my tests demonstrate that it already works on newer drivers). I even shared a link to the exact line of code showing the use of NVML.

I am just confused: I see many posts bashing NVIDIA, when support is already in the code. Maybe a leftover check in the GUI? I hope that someone figures it out sooner rather than later.

**Comment by HackTestes** [comment]
@Krovikan

>Your work is very commendable. But we will all agree that the complete configuration module is needed. That is, this:

Thank you! And I 100% agree with you, we would be on a we better situation if nvidia-settings worked out of the box. My solution is mostly to:

1. Provide a stop gap solution to the problem of controlling the fans (I got tired of waiting and I wanted a solution ASAP);
2. Provide a proof of concept (just to show that the API already works);
3. Provide some features that I find particularly useful (ones that I know nvidia-settings won't ever support).

**Comment by HackTestes** [comment]
Just for reference, the NVML function that controls fan speed through NVML was added October 12, 2022. See the commit: https://github.com/NVIDIA/nvidia-settings/commit/f213c7bddff91634e6c4d9681e8a9a1b9883db88

Search for "DeviceSetFanSpeed_v2".

**Comment by HackTestes** [comment]
@BlueGoliath
> Nice. I'm sure Linux's "many" programmers will be releasing a nice GUI GPU utility software any day now.

Lol. But jokes aside, someone might have made a GUI app with NVML already (tuxclocker): https://github.com/Lurkki14/tuxclocker

I say might, because I didn't test it or saw evidence that it works on Wayland (aside from a few issues).

**Comment by HackTestes** [comment]
On another note, I have found the link of the NVIDIA employee I was talking about (found by accident again): https://github.com/NVIDIA/open-gpu-kernel-modules/discussions/262#discussioncomment-2939265

The original comment I am referring to:
> I forgot to mention, I also saw in a GitHub issue a NVIDIA employee talking that they were planning to add many features to NVML, instead of relying on libXNVCtrl (and that was how I learned about the possible importance of the NVML API). I just didn't put this information on the first post, because I couldn't find the issue again (I lost the link).

**Comment by pallaswept** [comment]
> I say might, because I didn't test it or saw evidence that it works on Wayland (aside from a few issues).

Those few issues are the important part. Ironically, Tuxclocker is basically non-working _because_ it uses NVML.

https://github.com/Lurkki14/tuxclocker/issues/89
https://github.com/Lurkki14/tuxclocker/issues/65

CoolerControl is a nice fan control util with a nice GUI, that works on (x)wayland. In spite of interest from the author, NVML hasn't been implemented there yet, mostly because the rust lib isn't complete (and implementing NVML without a wrapper lib leads to the sort of problems tuxclocker hit). 

It isn't a really pressing issue as it's only needed if someone is running wayland without xwayland - and likely, not even then. I found that nvidia-settings demanded a valid xauthority, but actually didn't need a real valid token to an X session on the machine, it just needed a valid token. So you can generate a fake xauthority and nvidia-settings still works.

Still, it would be nice if it was working properly on wayland, and we could spend all day listing out more wayland support features we need... Or, should I say, all decade. It's been almost 4 years, and counting, so....

**Comment by pallaswept** [comment]
I addressed the design flaws but perhaps the subtlety was lost on you, I was trying to show some respect to the dev. At least they're _trying_ to release something for us all. I feel like they ought to be supported. If you don't suffer from the same skill issue, then you're in a position to help them.

To put my post more simply, and without the subtlety: In reference to the post above mine: No, tuxclocker isn't working. But fan control is the least of our worries. We have that already. 

We need a lot of stuff that doesn't exist yet. 

"Everything in NVML works fine" was never under discussion here, what is, is that "things we need are still missing from NVML or anywhere else".

**Comment by bentenjamin** [comment]
+1 to please add more settings! I can't use gsync on wayland because I have to toggle "force gsync on unverified monitor" in nvidia settings :( gsync significantly increases my 1% lows and something I really want to use

**Comment by kbrenneman** [comment]
GSync (or VRR in general) is something that the each individual compositor has to implement, not something that the NVIDIA driver can control.

The NVIDIA driver can do that in Xorg because it's got a server-side driver module which handles modesetting and scanout. In Wayland, the compositor controls all of that through drm-kms.

**Comment by bentenjamin** [comment]
> GSync (or VRR in general) is something that the each individual compositor has to implement, not something that the NVIDIA driver can control.
> 
> The NVIDIA driver can do that in Xorg because it's got a server-side driver module which handles modesetting and scanout. In Wayland, the compositor controls all of that through drm-kms.

Hmm okay I hear you. Maybe a wrong assumption on my side then.
There have been multiple reports of VRR working; I do understand that VRR is very finicky and I have tried to be meticulous in my setup:

- KDE Plasma 6.1.1
- driver 555.58.02
- single monitor
- displayport
- fullscreen
- screen tearing disabled

After trying these settings with no success I assume it is an nvidia driver issue as I had to manually enable the “Allow G-SYNC on monitor not validated as G-SYNC Compatible” on both windows and xorg for gsync to work. Am I incorrect and the nvidia wayland driver not use/need this config?

**Comment by Fxzzi** [comment]
VRR does work correctly for me on Hyprland (wayland) without any extra configuration over what was originally required to make the card work properly. You are right in needing only one monitor connected. With my 2nd monitor attached gsync / vrr fails to function.

**Comment by Neo2003** [comment]
I did write a QT application using nvml directly, will post in github in coming days.
We can control fans, overclock and everything implemented by nvidia with nvml.

**Comment by Neo2003** [comment]
Let me share the tools I created using QT6.
The goal was not to create a killer app, but provide enough as simple as possible code for anyone to understand how to work with NVML.
https://github.com/Neo2003/nVidia-NVML-QT5-6-tools
Have fun

**Comment by Neo2003** [comment]
> > Let me share the tools I created using QT6. The goal was not to create a killer app, but provide enough as simple as possible code for anyone to understand how to work with NVML. https://github.com/Neo2003/nVidia-NVML-QT5-6-tools Have fun
> 
> Nice segfault.

Segfault where? Did I miss an allocation?

**Comment by Denizantip** [comment]
it's 2024 and the problem still persists.

**Comment by Krovikan** [comment]
Maybe...

https://www.phoronix.com/news/NVIDIA-Wayland-Roadmap-2024

**Comment by vkukk** [comment]
I guess, this is message from nvidia that we all should not use their product.

**Comment by XLRUUU** [comment]
It would be nice if the digital vibrance setting was available on Wayland too.

**Comment by tordne** [comment]
I just want to give my 2 cents, 

I see many people complaining and saying they use X11 instead of wayland, this is unnecessary and causes more issues than solves problems. I've done the following below on all my systems, and I can use external monitors, without immense difficulty. 

So if you have installed your nvidia driver correctly , then just create the following file `/etc/profile.d/nvidia.sh`
Add the following lines to this file

`export __NV_PRIME_RENDER_OFFLOAD=1
export __GLX_VENDOR_LIBRARY_NAME=nvidia
`
reboot and now your system should be using your nvidia graphics card for everything.

you can verify in the terminal with the following codes what your card is doing.

`nvidia-smi dmon` here you can see if your clocks are adaptive when you open new applications, etc
`nvidia-smi`, but first open several applications and it should show every application you have opened at the bottom
`nvidia-smi -pm 1` this is persistence mode. this should be turned on, to minimise lag.

After years of people begging, don't rely on nvidia to suddenly change their ways.

Nonetheless Nvidia, please you make enough money to actually hire a few good developers to fix this wayland settings problem. Wayland is not going away, X11 is outdated or for minimalist systems, so please put a bit of effort and money in, please, we all like to see some  things in a GUI instead of using the CLI for everything.

**Comment by sergiotca** [comment]
> I just want to give my 2 cents,
> 
> I see many people complaining and saying they use X11 instead of wayland, this is unnecessary and causes more issues than solves problems. I've done the following below on all my systems, and I can use external monitors, without immense difficulty.
> 
> So if you have installed your nvidia driver correctly , then just create the following file `/etc/profile.d/nvidia.sh` Add the following lines to this file
> 
> `export __NV_PRIME_RENDER_OFFLOAD=1 export __GLX_VENDOR_LIBRARY_NAME=nvidia ` reboot and now your system should be using your nvidia graphics card for everything.
> 
> you can verify in the terminal with the following codes what your card is doing.
> 
> `nvidia-smi dmon` here you can see if your clocks are adaptive when you open new applications, etc `nvidia-smi`, but first open several applications and it should show every application you have opened at the bottom `nvidia-smi -pm 1` this is persistence mode. this should be turned on, to minimise lag.
> 
> After years of people begging, don't rely on nvidia to suddenly change their ways.
> 
> Nonetheless Nvidia, please you make enough money to actually hire a few good developers to fix this wayland settings problem. Wayland is not going away, X11 is outdated or for minimalist systems, so please put a bit of effort and money in, please, we all like to see some things in a GUI instead of using the CLI for everything.

I appreciate the candor, but just throwing environment variables at your system is not a solution for not having a proper GPU settings panel. Also these doesn't even apply for people not using hybrid graphics. Remember, not all systems are built equal. 

Honestly, Nvidia just needs to step up their game and allow Linux users to experience all the features we paid for.

**Comment by GenocideStomper** [comment]
> It would be nice if the digital vibrance setting was available on Wayland too.

Digital vibrance and sharpness without having to apply filters in post which add 1-2 frames of latency

**Comment by erkexzcx** [comment]
https://forums.developer.nvidia.com/t/artifacts-vertical-lines-in-qhd-resolution-over-displayport-connection/109965

It's been reported 5 years ago, and on Wayland there is nothing I can do to fix this...

**Comment by tom-jm69** [comment]
bump

**Comment by Nshtk** [comment]
I would like to see Wayland support for Nvidia settings too!

**Comment by Krovikan** [comment]
575 Beta driver and nothing still...

![Image](https://github.com/user-attachments/assets/e22d4f6a-b1ab-4a51-b6d7-304034dc2b52)

**Comment by tom-jm69** [comment]
https://aur.archlinux.org/packages/nvibrant-bin works on wayland.

**Comment by tom-jm69** [comment]
It is not a complete replacement, but it was the only thing that kept me from switching to Wayland.

**Comment by Tremeschin** [comment]
@haras-unicorn I mainly had to understand how to interface with [NVIDIA/open-gpu-kernel-modules](https://github.com/NVIDIA/open-gpu-kernel-modules/) instead of reversing what X11 does, but I wouldn't be surprised if it had ended up in the same `/dev/nvidia-modeset` file deep in the code! 

Yes, it's far from all the features of `nvidia-settings`, for that we should wait for [NVML](https://developer.nvidia.com/management-library-nvml) to mature. But until then, people have been successful changing settings such as [fan curves, power](https://github.com/HackTestes/NVML-GPU-Control), and probably clocks, voltages, etc (no warranty!)

Similar to @tom-jm69, it was also the only setting holding me on X, for some it was [dithering](https://github.com/NVIDIA/nvidia-settings/issues/106#issuecomment-2773640870). Regardless of that, there's a lot more [unexplored options](https://github.com/NVIDIA/open-gpu-kernel-modules/blob/e00332b05f433f1fe2e465d4741dfa7e41bf0a72/src/nvidia-modeset/interface/nvkms-api.h#L2699-L2723) in the open-gpu api structs easily replaceable in my code, and/or [stuff before](https://github.com/NVIDIA/open-gpu-kernel-modules/blob/e00332b05f433f1fe2e465d4741dfa7e41bf0a72/src/nvidia-modeset/interface/nvkms-api.h#L210-L277) the monitor attributes part related to the GPU. It's up to experimentation modifying it solving what you all miss on Wayland 🙂

**Comment by XurxoMF** [comment]
06/2025. Are there any news on this topic?

**Comment by pshirshov** [comment]
> > 06/2025. Are there any news on this topic?
> 
> Learn to code and fork it.

I guess nvidia engineers are paid for this job. I'm not.

**Comment by haras-unicorn** [comment]
please stop spamming - if u wanna argue go do it somewhere else

the rest of us subscribe to issues like this so we can contribute and look out for possible changes/fixes

not for arguments

**Comment by Ratione** [comment]
Still an Issue.

**Comment by BlueGoliath** [comment]
> Still an Issue.

please stop spamming - if u wanna spam go do it somewhere else

the rest of us subscribe to issues like this so we can contribute and look out for possible changes/fixes

not for spam

**Comment by pshirshov** [comment]
> we can contribute and look out for possible changes/fixes

Yeah, we all looking forward to contribute for free into the IP of the most valued company on the planet.

**Comment by haras-unicorn** [comment]
when i said contribute i meant contribute to the thread with possible workarounds/fixes

**Comment by pshirshov** [comment]
There literally are no workarounds apart from spawning X just to configure the GPU (the settings are being preserved when X stops). Nvidia must put their crap together and support Wayland.

**Comment by haras-unicorn** [comment]
well apparently nvibrant also works for some people

i feel like im just contributing to the spam by replying so ill stop here sorry

**Comment by BlueGoliath** [comment]
> > we can contribute and look out for possible changes/fixes
> 
> Yeah, we all looking forward to contribute for free into the IP of the most valued company on the planet.

The Linux community sure does has a lot of entitled sideline cheerleaders.

Anyway, please stop spamming - if u wanna spam go do it somewhere else

the rest of us subscribe to issues like this so we can contribute and look out for possible changes/fixes

not for spam
